dali/python/backend_impl.cc

```diff
@@ -188,21 +189,23 @@ void FillTensorFromDlPack(py::capsule capsule, SourceDataType<SrcBackend> *batch
   // shared ptr is destroyed
   auto typed_shape = ConvertShape(shape, batch);
   bool is_pinned = dl_tensor.device.device_type == kDLCUDAHost;
-  batch->ShareData(shared_ptr<void>(dl_tensor.data,
-                                    [dlm_tensor_ptr = move(dlm_tensor_ptr)](void*) {}),
-                                    bytes, is_pinned, typed_shape, dali_type.id());
-
+  int device_id = CPU_ONLY_DEVICE_ID;
   // according to the docs kDLCUDAHost = kDLCPU | kDLCUDA so test it as a the first option
   if (dl_tensor.device.device_type == kDLCUDAHost) {
-    batch->set_device_id(-1);
+    device_id = CPU_ONLY_DEVICE_ID;
   } else if (dl_tensor.device.device_type == kDLCPU) {
-    batch->set_device_id(-1);
+    device_id = CPU_ONLY_DEVICE_ID;
   } else if (dl_tensor.device.device_type == kDLCUDA) {
-    batch->set_device_id(dl_tensor.device.device_id);
+    device_id = dl_tensor.device.device_id;
   } else {
     DALI_FAIL(make_string("Not supported DLPack device type: ", dl_tensor.device.device_type, "."));
   }
 
+  batch->ShareData(shared_ptr<void>(dl_tensor.data,
```

Comment:
kind of nitpick, but could you move the comment about empty lambda from L188 to L204? It departed quite far from the original target.